snapcraft.yaml

```diff
@@ -136,6 +138,10 @@ parts:
     #source-branch: feature-snapcraft-build
     # We should specify either the source-branch or the source-tag, but not both
     # source-tag: R2019b
+    override-build: |
+      npm install -g npm@next   # Because the the default version in Ubuntu 18.04 is broken
+      ./scripts/install/linux_optional_compilation_dependencies.sh
```

Comment:
It's very good to use the `linux_optional_compilation_dependencies.sh` script!
Even better if it would allow us to get rid of the long list of build dependencies that we have to maintain.

Reply:
Yes, you could probably cut way back on the build-packages list if you install the right few things from the script. The trick is determining what those "right few things" are. :smiley:

Reply:
I started checking which dependencies should no longer be needed (most of them).
Tomorrow I will update the `linux_optional_compilation_dependencies.sh` and fix the list of dependencies that are still required.

Reply:
Also, I suspect that one downside of such an approach is that it might be more time-consuming to debug issues with building the snap. I *think* snapcraft checkpoints the build after installing the build packages so they don't need to be reinstalled if the build fails. If you move installation of those packages to the build step then they will need to be reinstalled every time you make a fix and need to rebuild. Calling the script from an override-pull scriptlet instead would probably help with that.

Reply:
What command line are you using to build the snap under Ubuntu? Is it just `snapcraft` or `snapcraft --debug`? Or are you using other options?

Reply:
Here is the command used to build the snap:
https://github.com/cyberbotics/webots-snap/blob/3ced9d75452de7477de124f4059dc3afc2f486ae/.travis.yml#L81

I will check the performance, but if this will only affect the debugging then we don't care and the advantages of not having to maintain the dependencies are bigger than the slowdown while debugging locally.

Reply:
I'd seen that in .travis.yml but thought you might be using something else locally during development. 

Fwiw, I suspect that the `--destructive-mode` flag might explain why you were able to build snaps but I couldn't under Fedora. I believe that allows the build to use packages that are installed on the host (in this case in the travis vm). Recommend getting rid of that flag to ensure this issue doesn't sneak back in.

Reply:
Thank you for the suggestion.
I will evaluate removing the `destructive-mode` flag in a separate PR.